packages/ra-core/src/controller/input/ReferenceArrayInputBase.spec.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { testDataProvider } from 'ra-core';
+import { Basic, WithError } from './ReferenceArrayInputBase.stories';
+
+describe('<ReferenceArrayInputBase>', () => {
+    afterEach(async () => {
+        // wait for the getManyAggregate batch to resolve
+        await waitFor(() => new Promise(resolve => setTimeout(resolve, 0)));
+    });
+
+    it('should pass down the error if any occurred', async () => {
+        jest.spyOn(console, 'error').mockImplementation(() => {});
+        render(<WithError />);
+        await waitFor(() => {
+            expect(screen.queryByText('Error: fetch error')).not.toBeNull();
+        });
+    });
+    it('should pass the correct resource down to child component', async () => {
+        render(<Basic />);
+        await screen.findByText('Selected tags: 1, 3');
+    });
```

Comment:
This test seems redundant with `should apply default values`, and doesn't seem to check the resource.
You should either fix it or remove it IMO.

Reply:
It is a bit redundant but it checks the resource as its in the text we look for: _Seleted **tags**_

Reply:
indeed, my bad!